File: src/java/org/javaturk/dp/ch02/elevator/DataOnlyElevatorTest.java
package org.javaturk.dp.ch02.elevator;

import org.javaturk.dp.ch02.elevator.domain.DataOnlyElevator;

public class DataOnlyElevatorTest {
	
	private static DataOnlyElevator publicElevator = new DataOnlyElevator();
	static boolean stopped = false;

	public static void main(String args[]) {
		
		printElevatorState();
		publicElevator.doorOpen = true; // passengers get on
		
		publicElevator.doorOpen = false; // doors closed
		printElevatorState();
		// go down to currentFloor 0 (below bottom of building)
		publicElevator.currentFloor--;
		publicElevator.currentFloor++;

		// jump to currentFloor 7 (only 5 floors in building)
		publicElevator.currentFloor = 7;
		publicElevator.doorOpen = true; // passengers get on/off
		
		printElevatorState();
		publicElevator.doorOpen = false;
		
		publicElevator.currentFloor = -15; // go to the first floor
		printElevatorState();
		publicElevator.doorOpen = true; // passengers get on/off
		publicElevator.currentFloor++; // elevator moves with door open
		printElevatorState();
		publicElevator.doorOpen = false;
		publicElevator.currentFloor--;
		publicElevator.currentFloor--;
		printElevatorState();
		publicElevator.currentFloor--;
		printElevatorState();
		
		System.out.println("***********************");
		
		Thread thread = new Thread(){
			public void run(){
				System.out.println("Started!");
				try {
					sleep(1);
				} catch (InterruptedException e) {
					// TODO Auto-generated catch block
					e.printStackTrace();
				}
				stopped = true;
				System.out.println("Stopped!");
			}
		};
		
		thread.start();
		
		while(!stopped){
			publicElevator.currentFloor--;
		}
		printElevatorState();
		
	}
	
	public static void printElevatorState(){
		String doorStatusString = "CLOSED";
		if(publicElevator.doorOpen)
			doorStatusString = "OPEN";
		
		System.out.println("Elevator is at floot " + publicElevator.currentFloor + " and the door is " + doorStatusString);
	}
}

File: src/c#/Elevator/DataOnlyElevatorTest.cs
using System;

namespace Elevator {
	public class DataOnlyElevatorTest {
		private static DataOnlyElevator PublicElevator = new DataOnlyElevator();

		public static void Main(string[] args) {

			PrintElevatorState();
			PublicElevator.doorOpen = true; // passengers get on

			PublicElevator.doorOpen = false; // doors closed
			PrintElevatorState();
			// go down to currentFloor 0 (below bottom of building)
			PublicElevator.currentFloor--;
			PublicElevator.currentFloor++;

			// jump to currentFloor 7 (only 5 floors in building)
			PublicElevator.currentFloor = 7;
			PublicElevator.doorOpen = true; // passengers get on/off

			PrintElevatorState();
			PublicElevator.doorOpen = false;

			PublicElevator.currentFloor = -15; // go to the first floor
			PrintElevatorState();
			PublicElevator.doorOpen = true; // passengers get on/off
			PublicElevator.currentFloor++; // elevator moves with door open
			PrintElevatorState();
			PublicElevator.doorOpen = false;
			PublicElevator.currentFloor--;
			PublicElevator.currentFloor--;
			PrintElevatorState();
			PublicElevator.currentFloor--;
			PrintElevatorState();


		}


	    public static void PrintElevatorState() {
		    String doorStatusString = "CLOSED";
		    if (PublicElevator.doorOpen)
			    doorStatusString = "OPEN";

		    Console.WriteLine("Elevator is at floot " + PublicElevator.currentFloor + " and the door is " + doorStatusString);
	    }
	}
}

